### PR TITLE
WIP: Json rpc improvements

### PR DESCRIPTION
**Pull request by notlesh**
This is a work in progress. The goal is to remove `libabyss` and replace it with `libjson-rpc-cpp`, restoring the original functionality along the way.

### CODE REVIEW

File: llarp/CMakeLists.txt [hunk 1]
```diff
@@ -61,7 +61,9 @@ if(ANDROID)
   set(LOG_LIB log)
 endif()
 
-target_link_libraries(${UTIL_LIB} PUBLIC ${CRYPTOGRAPHY_LIB} ${FS_LIB} ${LOG_LIB})
+set(JSONCPP_LIBS jsoncpp microhttpd jsonrpccpp-common jsonrpccpp-server)
```
Comment (majestrate) [inline llarp/CMakeLists.txt]: isn't microhttpd GPL? also we already have a json parser we shouldn't pull another one in

Comment (notlesh) [reply llarp/CMakeLists.txt]: `microhttpd` is LGPL 2.1, and optionally "eCos License" if HTTPS/TLS is disabled. From [here](https://www.gnu.org/software/libmicrohttpd/):


> Libmicrohttpd is free software; you can redistribute it and/or modify it under the GNU LGPL v2.1 or at your option any later version. If you disable HTTPS/TLS support, you can also choose the second license, the eCos License. If you have questions about licensing, please contact the maintainer.

Agreed on the JSON parser; mixing parsers is certainly not ideal. What I think the library buys us is flexibility in the transport mechanism (HTTP/HTTPS, TCP, socket, etc.) IMO, the rest of it is of little value; implementing JSON-RPC isn't tricky, and this library doesn't do a great job at that.

While I don't want to spend much time on this, I do want something that doesn't suck, so I'm up for suggestions.

**Comment by despair86** [comment]
As the guy maintaining the windows port this is already looking good

**Comment by despair86** [comment]
we could just nuke `JSON for Modern C++` altogether
it doesn't even optimise that well to begin with

File: llarp/router/router.hpp [hunk 2]
```diff
@@ -253,7 +253,6 @@ namespace llarp
     std::string rpcBindAddr = DefaultRPCBindAddr;
 
     /// lokid caller
-    std::unique_ptr< rpc::Caller > rpcCaller;
```
Comment (majestrate) [inline llarp/router/router.hpp]: we need this in order to talk with lokid for obtaining the service node list

**Pull request closed without merging**